src/test/java/com/wikia/webdriver/testcases/visualeditor/VEVideoTests.java

```diff
@@ -0,0 +1,176 @@
+package com.wikia.webdriver.testcases.visualeditor;
+
+import com.wikia.webdriver.common.contentpatterns.PageContent;
+import com.wikia.webdriver.common.contentpatterns.VideoContent;
+import com.wikia.webdriver.common.core.annotations.Execute;
+import com.wikia.webdriver.common.core.helpers.User;
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.pageobjectsfactory.componentobject.visualeditordialogs.VisualEditorAddMediaDialog;
+import com.wikia.webdriver.pageobjectsfactory.componentobject.visualeditordialogs.VisualEditorMediaSettingsDialog;
+import com.wikia.webdriver.pageobjectsfactory.componentobject.visualeditordialogs.VisualEditorSaveChangesDialog;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.article.ArticlePageObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.visualeditor.VisualEditorPageObject;
+
+import org.joda.time.DateTime;
+import org.openqa.selenium.Dimension;
+import com.wikia.webdriver.common.dataprovider.VisualEditorDataProvider.Setting;
+import com.wikia.webdriver.common.dataprovider.VisualEditorDataProvider.ImageSize;
+import org.testng.annotations.Test;
+
+public class VEVideoTests extends NewTestTemplate {
+  // AM01
+  @Test(groups = {"VEVideo", "VEAddExternalVideoTests_001", "VEAddExternalVideo",
+      "VEAddExternalVideoTests_004"})
+  @Execute(asUser = User.USER)
+  public void VEAddExternalVideoTests_001_AddNonPremiumVid() {
+    VisualEditorPageObject ve =
+        new VisualEditorPageObject(driver).openVEOnArticle(wikiURL, PageContent.ARTICLE_NAME_PREFIX
+            + DateTime.now().getMillis());
+    ve.verifyVEToolBarPresent();
+    ve.verifyEditorSurfacePresent();
+    VisualEditorAddMediaDialog mediaDialog = ve.clickVideoButton();
+    VisualEditorPageObject veNew = mediaDialog.addMediaByURL(VideoContent.NON_PREMIUM_VIDEO_URL);
+    veNew.verifyVideos(1);
+    veNew.verifyVEToolBarPresent();
+    VisualEditorSaveChangesDialog save = veNew.clickPublishButton();
+    ArticlePageObject article = save.savePage();
+    article.verifyVEPublishComplete();
+  }
+
+  //AM02
+  @Test(groups = {"VEVideo", "VEAddExternalVideoTests_002", "VEAddExternalVideo"})
+  @Execute(asUser = User.USER)
+  public void VEAddExternalVideoTests_002_AddPremiumVid() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)